src/platform/checksum.cpp

```diff
@@ -28,8 +28,26 @@ namespace platform {
         {
             f->ReadAll( data );
         }
+        std::string dataNoEol("");
```

Comment:
Probs I wasn't clear, I'd suggest removing lineendings altogether. UNIX-like lineending is \n, mac os before 9 uses \r, after that, \n.
Also consider using the boost string algorithms, erase_all / replace_all  http://www.boost.org/doc/libs/1_56_0/doc/html/string_algo/usage.html#idp432318416